iocore/hostdb/HostDB.cc

```diff
@@ -583,7 +468,7 @@ reply_to_cont(Continuation *cont, HostDBInfo *r, bool is_srv = false)
       ink_assert(!"missing hostname");
       cont->handleEvent(is_srv ? EVENT_SRV_LOOKUP : EVENT_HOST_DB_LOOKUP, NULL);
       Warning("bogus entry deleted from HostDB: missing hostname");
-      hostDB.delete_block(r);
+      hostDB.newCache->del(r->key);
```

Comment:
I have to agree with HRP - we should hold to STL conventions where possible, so this should be `erase`.